integration/spark/src/main/java/io/openlineage/spark/agent/lifecycle/plan/JdbcUrlSanitizerImpl.java

```diff
@@ -0,0 +1,69 @@
+package io.openlineage.spark.agent.lifecycle.plan;
+
+import com.google.common.collect.ImmutableSet;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Generic implementation of {@link JdbcUrlSanitizer} Removes username and password from jdbc url
+ */
+public class JdbcUrlSanitizerImpl implements JdbcUrlSanitizer {
```

Comment:
This is a lot of code for a couple of regex matches. I can pass all the tests with the following
```
    jdbcUrl = jdbcUrl.substring(5);
    return jdbcUrl.replaceAll(SLASH_DELIMITER_USER_PASSWORD_REGEX, "@")
            .replaceAll(COLON_DELIMITER_USER_PASSWORD_REGEX, "")
            .replaceAll("(?<=[?,;&:)=])\\(?(?i)(?:user|username|password)=[^;&,)]+(?:[;&;)]|$)", "");
```

Maybe we can just make this a single utility method and get rid of the interface and public class?